tests/end_to_end/CairoPrecompiles/test_cairo_precompiles.py

```diff
@@ -73,3 +74,13 @@ async def test_should_fail_precompile_caller_not_whitelisted(
             )
             with evm_error("CairoLib: call_contract failed"):
                 await cairo_counter_caller.incrementCairoCounter(max_fee=max_fee)
+
+        async def test_last_caller_address_should_be_eoa(
+            self, get_contract, cairo_counter_caller
+        ):
+            eoa = await get_eoa()
+            await cairo_counter_caller.incrementCairoCounter(caller_eoa=eoa)
+            last_caller_address = await cairo_counter_caller.getLastCaller(
+                caller_eoa=eoa
+            )
```

Comment:
```suggestion
            last_caller_address = await cairo_counter_caller.getLastCaller()
```

Reply:
I also think it's better if it's _explicit_ in the text that the one that account that calls contract is `eoa`  and the the last_caller_address resolves to the same EOA. if you are reading this from outside it's explicit in one read, whereas otherwise you need to know that the fixture, if no explicit caller is defined, calls `get_eoa`, and returns the same account as the one we got right before...

Reply:
I think that you removed the wrong `caller_eoa`. You need it for the tx on the line above, but not here for the `view` call